cmd/crane/cmd/optimize.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2018 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"log"
+
+	"github.com/google/go-containerregistry/pkg/crane"
+	"github.com/spf13/cobra"
+)
+
+// NewCmdOptimize creates a new cobra.Command for the optimize subcommand.
+func NewCmdOptimize(options *[]crane.Option) *cobra.Command {
+	var files []string
+
+	cmd := &cobra.Command{
+		Use:     "optimize SRC DST",
+		Hidden:  true,
+		Aliases: []string{"opt"},
+		Short:   "Optimize a remote container image from src to dst",
+		Args:    cobra.ExactArgs(2),
```

Comment:
It would be nice if this allowed one arg and would just retag the optimized version, with an optional `-t` flag for the dst if you want to leave src alone.

Reply:
IDK how I feel about in-place optimization, but most of the places I want this for are things that aren't yet producing these images.